openwisp_notifications/handlers.py

```diff
@@ -192,35 +192,43 @@ def send_email_notification(sender, instance, created, **kwargs):
     if not (email_preference and instance.recipient.email and email_verified):
         return
 
-    try:
-        subject = instance.email_subject
-    except NotificationRenderException:
-        # Do not send email if notification is malformed.
-        return
-    url = instance.data.get('url', '') if instance.data else None
-    body_text = instance.email_message
-    if url:
-        target_url = url
-    elif instance.target:
-        target_url = instance.redirect_view_url
-    else:
-        target_url = None
-    if target_url:
-        body_text += _('\n\nFor more information see %(target_url)s.') % {
-            'target_url': target_url
-        }
-
-    send_email(
-        subject=subject,
-        body_text=body_text,
-        body_html=instance.email_message,
-        recipients=[instance.recipient.email],
-        extra_context={
-            'call_to_action_url': target_url,
-            'call_to_action_text': _('Find out more'),
+    recipient_id = instance.recipient.id
+    cache_key = f'email_batch_{recipient_id}'
+
+    cache_data = cache.get(
+        cache_key,
+        {
+            'last_email_sent_time': None,
+            'batch_scheduled': False,
+            'pks': [],
+            'email_id': instance.recipient.email,
         },
     )
 
+    if cache_data['last_email_sent_time'] and EMAIL_BATCH_INTERVAL > 0:
+        # Case 1: Batch email sending logic
+        if not cache_data['batch_scheduled']:
+            # Schedule batch email notification task if not already scheduled
+            tasks.batch_email_notification.apply_async(
+                (instance.recipient.id,), countdown=EMAIL_BATCH_INTERVAL
+            )
+            # Mark batch as scheduled to prevent duplicate scheduling
+            cache_data['batch_scheduled'] = True
+            cache_data['pks'] = [instance.id]
+            cache.set(cache_key, cache_data)
+        else:
+            # Add current instance ID to the list of IDs for batch
+            cache_data['pks'].append(instance.id)
+            cache.set(cache_key, cache_data)
+        return
+
+    # Case 2: Single email sending logic
+    # Update the last email sent time and cache the data
+    cache_data['last_email_sent_time'] = timezone.now()
+    cache.set(cache_key, cache_data, timeout=EMAIL_BATCH_INTERVAL)
```

Comment:
if `EMAIL_BATCH_INTERVAL` is zero, wouldn't this data be cached indefinitely? If that's the case, it's a problem.